src/rust/engine/options/src/lib.rs

```diff
@@ -200,20 +199,28 @@ pub enum Source {
 
 #[derive(Debug)]
 pub struct OptionValue<T> {
+    pub derivation: Option<Vec<(Source, T)>>,
+    // Scalar options are always set from a single source, so we provide that
```

Comment:
Just documenting why we have this here but not for list or dict-valued options. We could also get rid of it, but it is slightly useful in error messages.

Reply:
Correct - it's broken out here because derivation may not be set, and we were already using it prior in a handful of user-facing messages.